### ISSUE TITLE
run-task shouldn't fetch tags from the head_repo

### ISSUE THREAD

**Issue by jcristau** [taskcluster/taskgraph#105]
https://github.com/taskcluster/taskgraph/commit/a9a5fae3d2cb569f0225aff55fd9ea327f8a4456 made run-task fetch tags from the head repo.  In a PR, where the head and base repos are different, tags can also be different, and conflicting, causing the fetch to fail.
I don't know the reason this was added but maybe it would be enough to explicitly fetch tags from the base repo, as the only thing we should care about from the head repo is the commit?

**Comment by jcristau** [comment]
cc @escapewindow

**Comment by JohanLorenzo** [comment]
Additional context: https://github.com/taskcluster/taskgraph/commit/a9a5fae3d2cb569f0225aff55fd9ea327f8a4456 had never been an issue until I publish https://github.com/taskcluster/taskgraph/pull/93. I agree with with @jcristau, I'm unsure about the need to fetch tags from the head repo. I guess there is some context I don't have anymore. 

That said, I wonder if this requirement is still useful now that taskgraph fetches the `base_ref` and the `base_rev`.

**Comment by escapewindow** [comment]
I don't know why missing tags would have been an issue, unless we were explicitly setting "my head rev is TAG" for github-release tasks, which is certainly possible in the pre-taskgraph days. It's quite possible we don't need this anymore, and best practice would be to use revisions instead of tags.

**Comment by escapewindow** [comment]
Most likely we moved a focus-android tag to trigger this bustage -- in git, moving tags is not best practice. Still, not pulling tags is also good :)

**Comment by JohanLorenzo** [comment]
I agree. I don't have the full story but I wouldn't be surprised if someone indeed moved a tag. That said, we can't control what people do with their forks: 
 * they may create tags that will eventually exist on the `base_repository`,
 * then, they may open an unrelated PR
 * which breaks the tags pull anyway.

So I'd say better be safe than sorry and not pulling all tags is the way to go.
